ktor-network/windows/src/io/ktor/network/selector/SelectUtilsWindows.kt

```diff
@@ -138,19 +138,20 @@ internal actual class SelectorHelper {
         closeSet: MutableSet<Int>,
         completed: MutableSet<EventInfo>,
         wsaIndex: Int,
-        wsaEvents: List<COpaquePointer?>
+        wsaEvents: Map<Int, COpaquePointer?>
     ) {
+        println("proce")
         while (true) {
             val event = closeQueue.removeFirstOrNull() ?: break
             closeSet.add(event)
         }
 
-        watchSet.forEachIndexed { index, event ->
+        watchSet.forEach { event ->
             if (event.descriptor in closeSet) {
                 completed.add(event)
-                return@forEachIndexed
+                return@forEach
             }
-            val wsaEvent = wsaEvents[index]
```

Comment:
This caused the `IndexOutOfBoundsException` as the size of `wsaEvents` does not match the `watchSet` size when a single descriptor is added multiple times (when suspended on read and write). Changed it to a Map with the descriptor as key. Apologies for the issue!